src/libsyncengine/db/syncdb.cpp

```diff
@@ -1111,61 +984,44 @@ bool SyncDb::dbIds(std::unordered_set<DbNodeId> &ids, bool &found) {
 
-    bool parentNodeDbIdIsNull;
-    DbNodeId parentNodeDbId;
-    ASSERT(queryIsNullValue(SELECT_NODE_BY_NODEID_LITE_ID, SELECT_NODE_BY_NODEID_PARENTID, parentNodeDbIdIsNull));
-    if (!parentNodeDbIdIsNull) {
-        ASSERT(queryInt64Value(SELECT_NODE_BY_NODEID_LITE_ID, SELECT_NODE_BY_NODEID_PARENTID, parentNodeDbId));
-
-        SyncName localName;
-        ASSERT(querySyncNameValue(SELECT_NODE_BY_NODEID_LITE_ID, SELECT_NODE_BY_NODEID_NAMELOCAL, localName));
-        SyncName driveName;
-        ASSERT(querySyncNameValue(SELECT_NODE_BY_NODEID_LITE_ID, SELECT_NODE_BY_NODEID_NAMEDRIVE, driveName));
-        names.push_back({localName, driveName});
-
-        while (!parentNodeDbIdIsNull) {
-            ASSERT(queryResetAndClearBindings(SELECT_NODE_BY_NODEID_LITE_ID));
-            ASSERT(queryBindValue(SELECT_NODE_BY_NODEID_LITE_ID, 1, parentNodeDbId));
-            if (!queryNext(SELECT_NODE_BY_NODEID_LITE_ID, found)) {
-                LOG_WARN(_logger, "Error getting query result: " << SELECT_NODE_BY_NODEID_LITE_ID << " - nodeId=" << dbNodeId);
-                return false;
-            }
-            if (!found) {
-                return true;
-            }
+    std::string requestId = SELECT_ANCESTORS_NODES_REQUEST_ID;
 
-            ASSERT(queryIsNullValue(SELECT_NODE_BY_NODEID_LITE_ID, SELECT_NODE_BY_NODEID_PARENTID, parentNodeDbIdIsNull));
-            if (!parentNodeDbIdIsNull) {
-                ASSERT(queryInt64Value(SELECT_NODE_BY_NODEID_LITE_ID, SELECT_NODE_BY_NODEID_PARENTID, parentNodeDbId));
+    ASSERT(queryResetAndClearBindings(requestId));
+    ASSERT(queryBindValue(requestId, 1, dbNodeId));
 
-                ASSERT(querySyncNameValue(SELECT_NODE_BY_NODEID_LITE_ID, SELECT_NODE_BY_NODEID_NAMELOCAL, localName));
-                ASSERT(querySyncNameValue(SELECT_NODE_BY_NODEID_LITE_ID, SELECT_NODE_BY_NODEID_NAMEDRIVE, driveName));
-                names.push_back({localName, driveName});
-            }
+    for (;;) {
+        bool hasNext = false;
+        if (!queryNext(requestId, hasNext)) {
+            LOG_WARN(_logger, "Error getting query result: " << requestId.c_str());
+            return false;
         }
+        if (!hasNext) {
+            break;
+        }
+
+        SyncName nameLocal;
+        ASSERT(querySyncNameValue(requestId, 0, nameLocal));
+        SyncName nameDrive;
+        ASSERT(querySyncNameValue(requestId, 1, nameDrive));
+
+        names.push_back({nameLocal, nameDrive});
     }
 
-    ASSERT(queryResetAndClearBindings(SELECT_NODE_BY_NODEID_LITE_ID));
+    ASSERT(queryResetAndClearBindings(requestId));
 
-    // Construct path from names' vector
-    localPath.clear();
-    remotePath.clear();
-    for (auto nameIt = names.rbegin(); nameIt != names.rend(); ++nameIt) {
-        localPath.append(nameIt->first);
-        remotePath.append(nameIt->second);
+    found = (names.size() > 0);
+
+    if (found) {
```

Comment:
We may omit it because the for-loop below is empty if `found` is `false`. Besides, clearing the paths at the very start would be a more predictable of the function call.